tgui/packages/tgui/interfaces/space_heater.js

```diff
@@ -0,0 +1,274 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author IPingu (https://github.com/IPling)
+ * @license ISC
+ */
+
+// Notes: Change icons to the respective emag variants, give glitch affects to emag variants
+// Supercooling and Overheating emag affects.
+// Supercooling: Blue coloring to buttons and loading bar, picklocking affect for taking out the power cell.
+// Overheating: Red coloring to buttons and loading bar, power cell burns player when taken out.
+
+import { useBackend } from '../backend';
+import { Box, Button, Divider, Flex, Icon, LabeledList, Modal, ProgressBar, Section, Slider, Stack } from '../components';
+import { Window } from '../layouts';
+import { randInt } from './common/mathUtils';
+import { glitch } from './common/stringUtils';
+
+const Glitch_Text = (emagged, string, number) => {
+  if (emagged) {
+    return glitch(string, number);
+  } else {
+    return string;
+  }
+};
+// Used for well, finding the theme, but also to tell if its in the 'special variants' when emagged.
+const Find_Theme = (emagged, temperature, on) => {
+  if (!emagged) {
+    return "generic";
+  } else if (temperature < 100 && on) { // Under 100 kelvin is supercooling
+    return "ntos";
+  } else if (temperature > 400 && on) { // Over 400 kelvin is overheating
+    return "syndicate";
+  } else {
+    return "generic";
+  }
```

Comment:
you can write this a whole lot clearer if you aren't scared of one indent level

Reply:
Ah sure, if its really not clear enough.